### PR TITLE
Fix reused registry ignoring Prometheus options

### PR DESCRIPTION
**Pull request by jotak**
Also add test, example & documentation

### CODE REVIEW

**Comment by jotak** [comment]
Fixes #71 

@nolequen thank you for your PR, but as said, I think it's better to not add multiple different ways to inject an existing registry, unless there's good reasons for that. Does that sound good to you?

File: src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -279,7 +279,17 @@ As an example, here is a custom timer that will track the execution time of a pi
 For more examples, documentation about the Micrometer registry and how to create metrics, check
 link:http://micrometer.io/docs/concepts#_registry[Micrometer doc].
 
-=== Other instrumentation
+=== Reusing an existing registry
+
+It is possible to reuse an existing Micrometer registry (or `CollectorRegistry` from the Prometheus java client),
```
Comment (tsegismont) [inline src/main/asciidoc/index.adoc]:
```suggestion
It is possible to reuse an existing Micrometer registry (or `CollectorRegistry` from the Prometheus Java client),
```

File: src/main/java/examples/MicrometerMetricsExamples.java [hunk 2]
```diff
@@ -246,4 +248,22 @@ public DistributionStatisticConfig configure(Meter.Id id, DistributionStatisticC
           }
         });
   }
+
+  public void useExistingRegistry() {
+    // This registry might be used to collect metrics other than Vert.x ones
+    PrometheusMeterRegistry registry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+
+    // You could also reuse an existing registry from the Prometheus java client:
```
Comment (tsegismont) [inline src/main/java/examples/MicrometerMetricsExamples.java]:
```suggestion
    // You could also reuse an existing registry from the Prometheus Java client:
```

**Comment by tsegismont** [comment]
@jotak after it's merged, can you please cherry-pick the commit to `3.7` branch? Thank you

**Pull request merged**